test/e2e/network_policy.go

```diff
@@ -39,286 +39,255 @@ failed or succesfully connected as expected.
 */
 
 var _ = framework.KubeDescribe("NetworkPolicy", func() {
+	var service *v1.Service
+	var podServer *v1.Pod
 	f := framework.NewDefaultFramework("network-policy")
 
-	It("should support a 'default-deny' policy [Feature:NetworkPolicy]", func() {
-		ns := f.Namespace
-
-		By("Create a simple server.")
-		podServer, service := createServerPodAndService(f, ns, "server", []int{80})
-		defer cleanupServerPodAndService(f, podServer, service)
-		framework.Logf("Waiting for Server to come up.")
-		err := framework.WaitForPodRunningInNamespace(f.ClientSet, podServer)
-		Expect(err).NotTo(HaveOccurred())
-
-		// Create a pod with name 'client-can-connect', which should be able to communicate with server.
-		By("Creating client which will be able to contact the server since no policies are present.")
-		testCanConnect(f, ns, "client-can-connect", service, 80)
-
-		By("Creating a network policy denying all traffic.")
-		policy := &networking.NetworkPolicy{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "deny-all",
-			},
-			Spec: networking.NetworkPolicySpec{
-				PodSelector: metav1.LabelSelector{},
-				Ingress:     []networking.NetworkPolicyIngressRule{},
-			},
-		}
+	Context("Single-port servers", func() {
```

Comment:
Reorg looks generally good; I didn't look in extreme detail.

Do you really need the single-port servers/multi-port servers split? Couldn't you just always listen on both ports but just ignore port 81 in the tests where you don't care about it?